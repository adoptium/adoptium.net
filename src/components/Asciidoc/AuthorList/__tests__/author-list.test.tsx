import React from "react"
import { render } from "@testing-library/react"
import { describe, expect, it } from "vitest"
import AuthorsList from ".."

describe("AuthorsList component", () => {
  it("renders correctly", () => {
    const authors = ["test-author", "another-test-author"]
    const { container } = render(<AuthorsList authors={authors} />)
    expect(container).toMatchSnapshot()
  })

  it("returns null when no authors are passed", () => {
    const { container } = render(<AuthorsList authors={[]} />)
    expect(container.firstChild).toBeNull()
  })
})
