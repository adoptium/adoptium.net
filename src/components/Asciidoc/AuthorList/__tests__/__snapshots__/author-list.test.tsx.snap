// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthorsList component > renders correctly 1`] = `
<div>
  <div
    class="mt-10 flex-col justify-center items-start gap-4"
  >
    <div
      class="self-stretch text-white my-5 text-lg font-semibold leading-normal"
    >
      Text
    </div>
    <div
      class="flex flex-wrap justify-start items-center space-x-[-12px]"
    >
      <a
        aria-label="test-author"
        href="https://github.com/test-author"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="test-author"
          class="rounded-full border-solid border-4 border-purple mb-6"
          data-nimg="1"
          decoding="async"
          height="55"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D55&w=128&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D55&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Ftest-author.png%3Fsize%3D55&w=128&q=75 2x"
          style="color: transparent;"
          width="55"
        />
      </a>
      <a
        aria-label="another-test-author"
        href="https://github.com/another-test-author"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="another-test-author"
          class="rounded-full border-solid border-4 border-purple mb-6"
          data-nimg="1"
          decoding="async"
          height="55"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fgithub.com%2Fanother-test-author.png%3Fsize%3D55&w=128&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fgithub.com%2Fanother-test-author.png%3Fsize%3D55&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fgithub.com%2Fanother-test-author.png%3Fsize%3D55&w=128&q=75 2x"
          style="color: transparent;"
          width="55"
        />
      </a>
    </div>
  </div>
</div>
`;
